README.md

```diff
@@ -0,0 +1,65 @@
+# AblyD
+
+This is a basic demo which takes [websocketd](https://github.com/joewalnes/websocketd), which makes command-line commands on a device available to other devices via WebSockets, and provides this functionality through [Ably](https://www.ably.com).
+
+## Setup
+
+Firstly you need to get an Ably API key. You can sign up for an account with [Ably](https://www.ably.com/) and access your API key from the [app dashboard](https://www.ably.com/accounts/any/apps/any/app_keys). 
+
+Now all you need to do is run the main go file, passing in the command line command you want to be run as a parameter! For example, to use a bash file `count.sh`, which is included in this repo's `examples` folder, just run:
+
+```bash
+~ $ ./ablyD --apikey=YOUR_API_KEY ./examples/bash/count.sh
+```
+
+The program is now running, waiting for a message in the `command` channel in Ably to be sent. The message's data field should match the structure, with the value `start` for the message's name:
+
+```json
+{
+  "MessageID": "unique string value",
+  "Args": [ "some", "additional", "args", "for", "the", "programs" ]
+}
+```
+
+Once the server receives a message in the `command` channel, it will start up an instance of the program, using the `Args` you specify in the message as well. Once the program has started up, the server will send a message onto the `command` channel of structure:
+
+```json
+{
+	"MessageID": "unique string value matching the requesting MessageID",
+	"Pid": "process_id",
+	"Namespace": "ablyd",
+	"ChannelPrefix": "ablyd:server_id"
+}
+```
+
+The client can identify the process which has started for them by the `MessageID`, then use the `Prefix` to connect to an input and an output channel for the process. These will be of structure `{Prefix}{Pid}:serverinput` and `{Prefix}{Pid}:serveroutput`.
+
+Subscribing to the `serveroutput` channel will allow the client to receive any stdout messages from the server. The client can also publish messages into the `serverinput` channel which will be passed into the stdin of the process.
+
+This will continue until the program naturally terminates, resulting in the process dying, or the client submits a message to the `serverinput` with data `KILL`.
```

Comment:
We should consider adding a third `signal` channel the client can use to send OS signals to the process rather than trying to re-purpose the stdin channel.

That being said, we should also consider just using a single channel with typed messages, rather than using different channels for stdin / stdout / signals. For example, there could be a single "data" channel with messages like `{"type":"STDIN","data":"..."}` and `{"type":"SIGNAL","data":"15"}`. This is pretty much what the SSH protocol does (see https://datatracker.ietf.org/doc/html/rfc4254#section-5), but is probably out of scope for what you're trying to achieve here 🙂